Block order fixes and PoF disabling

klercker opened the pull request:
- Fix bug coming from block order swap in PoS
- Disable PoF and PogF in PoS

klercker committed:
- Fix bug coming from block order swap in PoS
- Disable PoF and PogF in PoS

klercker commented:
@hanssv could you take a look at the `update_fraud_info()` code? it kind-of snuck in there and I'm a little bit hesitant on it?

klercker commented on apps/aecore/src/aec_chain_state.erl:
```diff
@@ -834,14 +839,16 @@ apply_and_store_state_trees(Node, PrevNode, TreesIn, ForkInfoIn, State) ->
     {DifficultyOut, Events}.
 
 update_fraud_info(ForkInfoIn, Node, State) ->
```
@hanssv could you take a look at the update_fraud_info() code? it kind-of snuck in there and I'm a little bit hesitant on it?

hanssv replied:
Looks ok...

hanssv reviewed (commented):
I've written half of this, so no I will not review it.

klercker commented:
Already tagged for april, but let's get this merged anyhoo.

venimus committed:
Add unit test coverage for PoS-aware fraud/sibling-block dispatch

Cover the 5 functions changed by this PR (maybe_add_pof/2,
update_fraud_info/3, db_sibling_blocks/1, maybe_pof/3, maybe_pogf/2) with
EUnit tests exercising both the new PoS branch (always no_fraud / swapped
key-micro height offsets / fraud-flag passthrough) and the original PoW
branch, to guard the previous unfixed bug against regression.

Exports the 5 functions under the existing -ifdef(TEST) block in
aec_chain_state.erl so they can be called directly from
aec_chain_state_tests.erl.
